src/internal_events/datadog_metrics.rs

```diff
@@ -14,9 +14,10 @@ impl InternalEvent for DatadogMetricsEncodingError {
     fn emit(self) {
         error!(
             message = "Failed to encode Datadog metrics.",
-            error = %self.message,
```

Comment:
I don't think this is quite right. I think this error log should stay as-is to match https://github.com/vectordotdev/vector/blob/master/docs/specs/instrumentation.md#error and then an _additional_ log should be emitted below which matches https://github.com/vectordotdev/vector/blob/master/docs/specs/instrumentation.md#eventsdropped.

You could refactor the below `if self.dropped_events > 0 {` to, rather than emitting the logs and metrics directly, instead emit a new event: `ComponentEventsDropped`, that would emit the requisite logs and metrics. This event could be re-used across places where we drop events.

Reply:
Whoa, yeah I was way off there... my bad!

Reply:
In this particular case , it looks like "error_code", "error_type" and "stage" are included here for the"component_discarded_events_total" metric. Though they are not required by the instrumentation spec. In creating a generic `ComponentEventsDropped` , I was going to just use what is required in the instrumentation spec. It seems like it might be convoluted to provide the means to add additional, optional tags in `ComponentEventsDropped` . Is it ok drop "error_code", "error_type" and "stage" in this metric?

Reply:
Attempted to address here https://github.com/vectordotdev/vector/pull/14052/commits/138b1abbc17c16382d00f130e1f2d325da4f1135